express-post-json/server.js

```diff
@@ -0,0 +1,34 @@
+import express from 'express';
+
+const port = 8080;
+const app = express();
+const parser = express.json();
+app.use(parser);
+
+let nextId = 1;
+const grades = {};
+
+app.get('/api/grades', (req, res) => {
+  const gradeArray = [];
+  for (const grade in grades) {
+    gradeArray.push(grades[grade]);
+  }
+  res.json(gradeArray);
+});
+
+app.post('/api/grades', (req, res) => {
+  if (req.body) {
+    const body = req.body;
+    body.id = nextId;
+    nextId++;
+    grades[body.id] = body;
+    res.status(201).send(body);
+  } else {
+    console.log('Unable to retrieve json information...');
+    res.sendStatus(404);
```

Comment:
I like the error handling! Note, though, that you should just send a 400 since the problem here is not "Not Found", but it's an incomplete request.